src/NServiceBus.Gateway/Deduplication/LegacyDeduplicationWrapper.cs

```diff
@@ -14,18 +14,32 @@ public LegacyDeduplicationWrapper(IDeduplicateMessages legacyPersister)
 
         public bool SupportsDistributedTransactions { get; } = true;
 
-        public async Task<bool> IsDuplicate(string messageId, ContextBag context)
+        public async Task<IDuplicationCheckSession> IsDuplicate(string messageId, ContextBag context)
```

Comment:
The method should be something else like `CreateDuplicateSessionForMessageId` or something.

Reply:
yeah I didn't spend time to rethink the method names on the spike. I agree that we should change this name though 👍